src/format.rs

```diff
@@ -9,16 +9,10 @@ pub fn key_with_ts(key: impl Into<BytesMut>, ts: u64) -> Bytes {
 
 pub fn append_ts(key: &mut BytesMut, ts: u64) {
     key.reserve(8);
-    let res = (u64::MAX - ts).to_be();
-    let buf = key.bytes_mut();
-    unsafe {
-        ptr::copy_nonoverlapping(
-            &res as *const u64 as *const u8,
-            buf.as_mut_ptr() as *mut _,
-            8,
-        );
-        key.advance_mut(8);
-    }
+    let mut ts_bytes = key.split_off(key.len());
+    let res = u64::MAX - ts;
+    ts_bytes.put_u64(res);
+    key.unsplit(ts_bytes);
 }
```

Comment:
Here I rewrote `append_ts` and `get_ts` in a safe approach. Benchmark result shows that safe version is nearly the same performance as previous unsafe version, and safe `get_ts` is 2x slower than unsafe version. So I retained previous `get_ts` and put safe `append_ts` here.

<img width="581" alt="image" src="https://user-images.githubusercontent.com/4198311/95561178-ede60880-0a4c-11eb-816a-33754bf3376e.png">

Reply:
I think there are problems with criterion or the compiler. I can get an opposite result after moving the legacy case as the first case to bench.